docs/.vitepress/theme/components/DiscordModal.vue

```diff
@@ -0,0 +1,70 @@
+<script setup lang="ts">
+import { onBeforeUnmount, onMounted, ref } from 'vue'
+
+const ONE_DAY_MS = 1000 * 60 * 60 * 24
+const STORAGE_KEY = 'discord-modal-last-opened'
+
+const isVisible = ref(false)
+
+const close = () => {
+  isVisible.value  = false
+  localStorage.setItem(STORAGE_KEY, String(new Date().getTime()))
+}
+
+onMounted(() => {
+  // get the last time we showed the modal, or 1 day ago if it's not set yet
+  const lastOpenedAt = localStorage.getItem(STORAGE_KEY) || (new Date().getTime()) - ONE_DAY_MS
+  const lastOpenDiff = new Date().getTime() - new Date(Number(lastOpenedAt)).getTime()
```

Comment:
Quite a few `Date` object created. Perhaps, it is possible to achieve the same without this many.

Reply:
Sure can do :cattyping: